pumpp/task/key.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python
+# -*- encoding: utf-8 -*-
+'''Key recognition task transformer'''
+
+import numpy as np
+import mir_eval
+import jams
+
+from librosa import note_to_midi
+
+from .base import BaseTaskTransformer
+from ..labels import MultiLabelBinarizer
+
+__all__ = ['KeyTransformer']
+
+class KeyTransformer(BaseTaskTransformer):
+    '''Key annotation transformer.
+
+    This transformer uses a (pitch_profile, tonic) decomposition of key_mode 
+    annotation, where the mode is reflected in the 12-D pitch_profile vector.
+
+    Attributes
+    ----------
+    name : str
+        The name of the key trnsformer
+
+    sr : number > 0
+        The sampling rate of audio
+
+    hop_length : int > 0
+        The number of samples between each annotation frame
+
+    sparse : bool
+        If True, tonic value is sparsely encoded as integers in [0, 12].
+        If False, tonic value is densely encoded as 13-dimensional booleans.
+    '''
+    def __init__(self, name='key', sr=22050, hop_length=512, sparse=False):
+        '''Initialize a key task transformer'''
+
+        super(KeyTransformer, self).__init__(name=name,
+                                             namespace='key_mode',
+                                             sr=sr, hop_length=hop_length)
+
+        self.encoder = MultiLabelBinarizer()
+        self.encoder.fit([list(range(12))])
```

Comment:
This will learn an encoder of [0, 11] ==> [0, 11].  Did you mean to have notes here instead?

Reply:
This is me copy-pasting from the chord.py module... 

Turns out I never used `KeyTransformer.encoder` for anything...
I should get rid of line 44-46

Reply:
I think there should be an encoder for the `tonic` field that maps `[C, C#, ..., B, N]` to target indices, and it will get used in the inverse transform.

Reply:
Tried to use the classes in `label.py`, but the default LabelTransformer will sort the labels according to the alphabet, which for note names is unfortunate... See below.
```
In [3]: tran._classes                                                                                                                                                                                                               
Out[3]: {'A', 'A#', 'B', 'C', 'C#', 'D', 'D#', 'E', 'F', 'F#', 'G', 'G#', 'N'}

In [4]: tran.encoder.transform(['C', 'D', 'E'])                                                                                                                                                                                     
Out[4]: 
array([[0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0]])
```
I think I'll deal with encoding decoding with my own functions first to implement inverse transform. If later I discover a good use for the sklearn label encoder classes I'll use them.

Reply:
Yeah, fair point.  I'm convinced that we don't need a label transformer for tonic.